src/app/public/modules/action-buttons/action-buttons.component.ts

```diff
@@ -44,18 +91,35 @@ export class StacheActionButtonsComponent implements OnInit {
 
   public searchApplied(searchText: string) {
     this.searchText = searchText;
-    this.filteredRoutes = this.routes;
     const query = searchText.toLowerCase();
 
+    // If search text is empty, reset array while honoring restricted routes.
     if (!searchText) {
+      this.filteredRoutes = this.filterRestrictedRoutes(this.routes, this.isAuthenticated);
       return;
     }
     this.filteredRoutes = this.routes.filter((route: any) => {
+
+      // Remove restricted routes from results if user is not a authenticated BB user.
+      if (route.restricted && !this.isAuthenticated) {
+        return false;
```

Comment:
doesn't `filterRestrictedRoutes` already do this?

Reply:
ooo i c. you call `filterRestrictedRoutes` in a diff branch of the if. 

🤔 it looks like no matter what `searchText` is, you want to filter out restricted right? so would this clean this up:

```typescript
     this.filteredRoutes = this.filterRestrictedRoutes(this.routes, this.isAuthenticated);
     if (!searchText) {
            this.filteredRoutes = this.filteredRoutes.filter(theMatchingFieldsLogic);
     }
```

hmmm, i suppose that would have implication on the change detection huh? maybe this

```typescript

     this.filteredRoutes = this.filterRestrictedRoutes(this.routes, this.isAuthenticated)
         .filter(x => theMatchingFieldsLogic(x, searchText);

function theMatchingFieldsLogic(ele): boolean {
  if (!searchText) return false;
  else {
      // the rest of the logic
  }
}

Reply:
That reads a lot better. Great suggestion. I've added your feedback to the PR.